lib/upipe/ubuf_pic_common.c

```diff
@@ -288,6 +288,53 @@ int ubuf_pic_common_check_skip(struct ubuf_mgr *mgr, int hskip, int vskip)
     return UBASE_ERR_NONE;
 }
 
+/** @This splits an interlaced picture ubuf in its two fields.
+ *
+ * Two extra ubufs are allocated, one per field.
+ *
+ * @param ubuf pointer to ubuf
+ * @param odd pointer to pointer to odd field ubuf
+ * @param even pointer to pointer to even field ubuf
+ * @return an error code
+*/
+int ubuf_pic_common_split_fields(struct ubuf *ubuf, struct ubuf **odd,
+        struct ubuf **even)
+{
+    *odd = ubuf_dup(ubuf);
+    if (!*odd)
+        return UBASE_ERR_ALLOC;
+
+    *even = ubuf_dup(ubuf);
+    if (!*odd) {
+        ubuf_free(*odd);
```

Comment:
Do you need to do anything about the value stored in odd?  Do you need to set it to NULL or anything?

Reply:
I guess he meant if (!even)

Reply:
No if the function returns an error the contents are undefined

Reply:
@cmassiot nice catch, fixed it